src/microbe_stage/systems/EngulfingSystem.cs

```diff
@@ -1629,8 +1627,8 @@ private void PerformEjectionForceAndAttachedRemove(in Entity entity, ref Engulfa
         physics.VelocitiesApplied = false;
 
         // Reset engulfable state after the ejection (but before RemoveEngulfedObject to allow this to still see
-        // the hostile engulfer entity)
-        engulfable.OnExpelledFromEngulfment(engulfableObject, spawnSystem, worldSimulation);
+        // the hostile engulfer entity). Delayed a frame to avoid "zombies"
+        engulfable.HandleEjectionFlag = true;
```

Comment:
This seems like a massive hack workaround around the real bug, which is likely inside OnExpelledFromEngulfment or how the death system sees the resulting ejected cell. As such I won't just accept these changes without understanding the real problem.